calyx/src/lang/structure.rs

```diff
@@ -405,6 +417,60 @@ impl StructureGraph {
         }
     }
 
+    /// Construct and insert an edge given two node indices
+    pub fn remove_edge<S: AsRef<str>, U: AsRef<str>>(
+        &mut self,
+        src_node: NodeIndex,
+        src_port: S,
+        dest_node: NodeIndex,
+        dest_port: U,
+    ) -> Result<(), Error> {
+        let src_port: &str = src_port.as_ref();
+        let dest_port: &str = dest_port.as_ref();
+        let find_width =
+            |port_to_find: &str, portdefs: &[ast::Portdef]| match portdefs
+                .iter()
+                .find(|x| &x.name == port_to_find)
+            {
+                Some(port) => Ok(port.width),
+                None => Err(Error::UndefinedPort(port_to_find.to_string())),
+            };
+        use NodeData::{Input, Instance, Output};
+        let src_width = match &self.graph[src_node] {
+            Instance { signature, .. } => {
+                find_width(src_port, &signature.outputs)
+            }
+            Input(portdef) => Ok(portdef.width),
+            Output(_) => Err(Error::UndefinedPort(src_port.to_string())),
+        }?;
+        let dest_width = match &self.graph[dest_node] {
+            Instance { signature, .. } => {
+                find_width(dest_port, &signature.inputs)
+            }
+            Input(_) => Err(Error::UndefinedPort(dest_port.to_string())),
+            Output(portdef) => Ok(portdef.width),
+        }?;
+        // if widths match, add edge to the graph
+        if src_width == dest_width {
+            match self.graph.find_edge(src_node, dest_node) {
+                Some(edge) => {
+                    self.graph.remove_edge(edge);
+                    Ok(())
+                }
+                None => Err(Error::UndefinedEdge(
+                    src_port.to_string(),
+                    dest_port.to_string(),
+                )),
+            }
+        } else {
+            Err(Error::MismatchedPortWidths(
+                self.construct_port(src_node, src_port),
+                src_width,
+                self.construct_port(dest_node, dest_port),
+                dest_width,
+            ))
+        }
+    }
```

Comment:
add a line between functions

Reply:
I am not sure which line are you referring...

Reply:
Add an empty line after the end of a function.